iclib/ltc6810.py

```diff
@@ -0,0 +1,622 @@
+        :return: ``None``.
+        """
+        command = 0b01001100000
+        command |= ch_mode.mode << 7
+        command |= DCP << 4
+        command |= CH
+
+        if address is None:
+            raise NotImplementedError
+        else:
+            transmitted_bytes = self.get_address_poll_command_bytes(
+                address,
+                command,
+                0,
+            )
+
+        self.spi.transfer(transmitted_bytes)
+
+        if sleep:
+            if CH:
+                timeout = ch_mode.cell_total_conversion_time
+            else:
+                timeout = ch_mode.all_cells_total_conversion_time
+
+            _sleep(timeout)
+
+    @dataclass
+    class CVAR:
+        """Cell voltage register group A."""
+
+        C1V: float
+        C2V: float
+        C3V: float
+
+    def RDCVA(self, address: int | None = None) -> CVAR:
+        """Read cell voltage register group A.
+
+        :param address: The optional address.
+        :return: ``None``.
+        """
+        if address is None:
+            raise NotImplementedError
+        else:
+            transmitted_bytes = self.get_address_read_command_bytes(
+                address,
+                0b00000000100,
+                6,
+            )
+
+        received_bytes = self.spi.transfer(transmitted_bytes)[-6:]
+
+        assert isinstance(received_bytes, list)
+
+        return self.CVAR(
+            self.get_voltage(received_bytes[0:2]),
+            self.get_voltage(received_bytes[2:4]),
+            self.get_voltage(received_bytes[4:6]),
+        )
+
+    @dataclass
+    class CVBR:
+        """Cell voltage register group B."""
+
+        C4V: float
+        C5V: float
+        C6V: float
+
+    def RDCVB(self, address: int | None = None) -> CVBR:
+        """Read cell voltage register group B.
+
+        :param address: The optional address.
+        :return: ``None``.
+        """
+        if address is None:
+            raise NotImplementedError
+        else:
+            transmitted_bytes = self.get_address_read_command_bytes(
+                address,
+                0b00000000110,
+                6,
+            )
+
+        received_bytes = self.spi.transfer(transmitted_bytes)[-6:]
+
+        assert isinstance(received_bytes, list)
+
+        return self.CVBR(
+            self.get_voltage(received_bytes[0:2]),
+            self.get_voltage(received_bytes[2:4]),
+            self.get_voltage(received_bytes[4:6]),
+        )
+
+    class CHGMode(Enum):
+        """The CHG ADC Modes, as defined in Page 63 of datasheet."""
+
+        M27000 = 0b01, 521e-6, 200e-6
+        """27 kHz Mode (Fast)."""
+        M14000 = 0b01, 695e-6, 229e-6
+        """14 kHz Mode."""
+        M7000 = 0b10, 1.2e-3, 403e-6
+        """7 kHz Mode (Normal)."""
+        M3000 = 0b10, 1.9e-3, 520e-6
+        """3 kHz Mode."""
+        M2000 = 0b11, 3.3e-3, 752e-6
+        """2 kHz Mode."""
+        M1000 = 0b00, 6.1e-3, 1.2e-3
```

Comment:
Datasheet says 6.0ms for 1khz, not 6.1ms

Reply:
Yup, good catch!